src/core/CMakeLists.txt

```diff
@@ -84,24 +87,83 @@ install(
   DESTINATION lib
 )
 
+################################################################################
+# tc_core_cuda_no_sdk
+#
+# Core CUDA cross-compilation library for compiling to CUDA and not running
+# Builds **without** the CUDA SDK and used in cross-compilation tests
+################################################################################
+add_library(
+  tc_core_cuda_no_sdk
+
+  SHARED
+
+  cuda/cuda_mapping_options.cc
+  cuda/cuda_mapping_options_cpp_printer.cc
+  polyhedral/cuda/codegen.cc
+  polyhedral/cuda/mapped_scop.cc
+  polyhedral/cuda/mapping_types.cc
+  polyhedral/cuda/memory_promotion_heuristic.cc
+  polyhedral/cuda/tighten_launch_bounds.cc
+)
+# Unsets CUDA_HOME to compile without SDK
+set_target_properties(tc_core_cuda_no_sdk
+                      PROPERTIES COMPILE_FLAGS
+                      "${CMAKE_CXX_FLAGS} -UCUDA_HOME -Wl,--no-undefined")
+target_include_directories(tc_core_cuda_no_sdk PUBLIC ${PROJECT_SOURCE_DIR}/include ${LLVM_INCLUDE_DIRS})
+target_link_libraries(
+  tc_core_cuda_no_sdk
+
+  ${HALIDE_LIBRARIES}
+  ${ISL_LIBRARIES}
+  -lLLVM
+
+  tc_lang
+  tc_version
+  tc_proto
+  tc_core
+)
+install(
+  TARGETS
+  tc_core_cuda_no_sdk
+
+  DESTINATION lib
+)
+
+################################################################################
+# tc_cuda
+#
+# Core CUDA library for compiling to PTX and running on actual GPUs
+################################################################################
 if (WITH_CUDA)
   find_library(CUDA_NVRTC_LIBRARIES nvrtc
     PATHS ${CUDA_TOOLKIT_ROOT_DIR}
     PATH_SUFFIXES lib lib64 targets/x86_64-linux/lib targets/x86_64-linux/lib/stubs)
 
   add_library(
-    tc_core_cuda
+    tc_cuda
 
     SHARED
 
+    # This block of file is copy-pasted from tc_core_cuda_no_sdk
+    # Don't know a better way atm
```

Comment:
I'm afraid I can't help here.  Not a cmake expert.

Reply:
declare a variable with this list and reuse it

Reply:
easy enough :) thanks!